pkgs/applications/science/molecular-dynamics/lammps/default.nix

```diff
@@ -1,54 +1,63 @@
 { stdenv, fetchFromGitHub
 , libpng, gzip, fftw, openblas
+, cmake, pkgconfig, python2
 , mpi ? null
 }:
-let packages = [
-     "asphere" "body" "class2" "colloid" "compress" "coreshell"
-     "dipole" "granular" "kspace" "manybody" "mc" "misc" "molecule"
-     "opt" "peri" "qeq" "replica" "rigid" "shock" "snap" "srd" "user-reaxc"
-    ];
-    lammps_includes = "-DLAMMPS_EXCEPTIONS -DLAMMPS_GZIP -DLAMMPS_MEMALIGN=64";
-    withMPI = (mpi != null);
+let
+  withMPI = (mpi != null);
+
+  packages = [
+    "asphere" "body" "class2" "colloid" "compress" "coreshell" "dipole"
+    "granular" "kspace" "manybody" "mc" "meam" "misc" "molecule" "opt"
+    "peri" "poems" "qeq" "replica" "rigid" "shock" "snap" "srd" "user-atc"
+    "user-fep" "user-manifold" "user-misc" "user-molfile" "user-reaxc"
+    "kokkos" "python"
+  ]
+  ++ stdenv.lib.optionals withMPI [ "mpiio" ];
+
+  lammps_includes = [
+    "-DLAMMPS_EXCEPTIONS=yes"
+    "-DLAMMPS_GZIP=yes"
+    "-DLAMMPS_MEMALIGN=64"
+    "-DBUILD_LIB=yes"
+  ];
+
 in
 stdenv.mkDerivation rec {
   # LAMMPS has weird versioning converted to ISO 8601 format
-  version = "stable_22Aug2018";
+  version = "stable_7Aug2019";
   pname = "lammps";
 
   src = fetchFromGitHub {
     owner = "lammps";
     repo = "lammps";
     rev = version;
-    sha256 = "1dlifm9wm1jcw2zwal3fnzzl41ng08c7v48w6hx2mz84zljg1nsj";
+    sha256 = "0yddiyizmqip87jvg76pqa8g3nk4sm0xwzryp9bq276q20l6mamj";
   };
 
   passthru = {
     inherit mpi;
     inherit packages;
   };
 
-  buildInputs = [ fftw libpng openblas gzip ]
-    ++ (stdenv.lib.optionals withMPI [ mpi ]);
-
-  configurePhase = ''
-    cd src
-    for pack in ${stdenv.lib.concatStringsSep " " packages}; do make "yes-$pack" SHELL=$SHELL; done
+  patchPhase = ''
+    substituteInPlace "cmake/CMakeLists.txt" --replace \
+      "$ENV{HOME}/.local" \
+      "$out"
   '';
 
-  # Must do manual build due to LAMMPS requiring a seperate build for
-  # the libraries and executable. Also non-typical make script
-  buildPhase = ''
-    make mode=exe ${if withMPI then "mpi" else "serial"} SHELL=$SHELL LMP_INC="${lammps_includes}" FFT_PATH=-DFFT_FFTW3 FFT_LIB=-lfftw3 JPG_LIB=-lpng
-    make mode=shlib ${if withMPI then "mpi" else "serial"} SHELL=$SHELL LMP_INC="${lammps_includes}" FFT_PATH=-DFFT_FFTW3 FFT_LIB=-lfftw3 JPG_LIB=-lpng
-  '';
+  nativeBuildInputs = [ cmake pkgconfig ];
+  buildInputs = [ fftw libpng openblas gzip python2 ]
+    ++ (stdenv.lib.optionals withMPI [ mpi ]);
 
-  installPhase = ''
-    mkdir -p $out/bin $out/include $out/lib
+  cmakeFlags = with stdenv;
+  map (pkg: "-DPKG_${lib.toUpper pkg}=yes") packages
+  ++ lib.optionals withMPI [ "-DBUILD_MPI=yes" ]
```

Comment:
Continuing the idea of putting `mpi` in the packages list, perhaps here, you can check if `"mpi"` exists in `packages` and add this cmake flag accordingly.